main/lib/core/src/FileDeserializer.cc

```diff
@@ -55,22 +56,33 @@ namespace eudaq {
       }
     }
     size_t read =
-        fread(reinterpret_cast<char *>(m_stop), 1, end - m_stop, m_file);
+      fread(reinterpret_cast<char *>(m_stop), sizeof(char), end - m_stop, m_file);
     m_stop += read;
+    int n_tries = 0;
+    const int max_tries = 1000;    
     while (read < min) {
       if (feof(m_file) && m_faileof) {
-        throw FileReadException("End of File encountered");
-      } else if (int err = ferror(m_file)) {
+        throw FileReadException("End of file '"+m_filename+"' encountered");
+      }
+      int errcode = ferror(m_file);
+      if (errcode!=0) {
         EUDAQ_THROWX(FileReadException,
-                     "Error reading from file: " + to_string(err));
-      } else if (m_interrupting) {
+                     "Error reading from file '"+m_filename+"': " + to_string(errcode));
+      }
+      if (m_interrupting) {
         m_interrupting = false;
         throw InterruptedException();
       }
+      if(n_tries >= max_tries){
```

Comment:
why do we need 1k attempts to read from a file?

Reply:
we don't. there was no limit before. we can put any number here you want.